process_log: Add CEL filtering

vitkyrka opened the pull request:
### What does this PR do?

Allow processes to be excluded from process log discovery using CEL by hooking it up to the [unified workload filtering component](https://datadoghq.atlassian.net/wiki/spaces/CONTP/pages/5050598441).

Example:
```
cel_workload_exclude:
  - products: ["logs"]
    rules:
      processes:
        - "process.name == 'nginx' && process.log_file.contains('access.log')"
  - products: ["global"]
    rules:
      processes:
        - "process.name == 'apache'"
```

### Motivation

https://datadoghq.atlassian.net/browse/DSCVR-209

### Describe how you validated your changes

Tests added.

vitkyrka committed:
process_log: Add CEL filtering

Allow process to be excluded from process log discovery using CEL.

vitkyrka committed:
Build CEL test under cel flag

gabedos commented on comp/core/workloadfilter/def/types.go:
```diff
@@ -385,4 +402,5 @@ type ProcessFilter int
 // Defined Process filter kinds.
 const (
 	LegacyProcessExcludeList ProcessFilter = iota
+	ProcessCELLogs
```
nit: To provide a bit of a more consistent experience across all of the other workload types, can we also define a `ProcessCELGlobal` filter definition?

Now it won't be that useful bc logs is the only product, but eventually we will want to configure these filters for process metrics and the global level rule would be more consistent with all the other definitions.

vitkyrka committed:
Use existing CreateProcess function

vitkyrka committed:
Remove nil check for workloadfilter component

vitkyrka committed:
Add global filter for processes

vitkyrka committed:
Merge remote-tracking branch 'origin/main' into vitkyrka/discolog-filter

s-alad reviewed (approved):
lgtm for agent-config files

vitkyrka committed:
Merge branch 'main' into vitkyrka/discolog-filter

vitkyrka committed:
process_log: Add filtering on log files (#42141)

### What does this PR do?

Allow filtering discovered log files by paths of the log files. For this
add a new `log_file` field to the Process type and call the filters once
for each log file.

Example:
```
cel_workload_exclude:
  - products: ["logs"]
    rules:
      processes:
        - "process.name == 'nginx' && process.log_file.contains('error.log')"
```

### Motivation

https://datadoghq.atlassian.net/browse/DSCVR-209

### Describe how you validated your changes

Tests added.

vitkyrka committed:
Add comment about log_file

vitkyrka committed:
Merge remote-tracking branch 'origin/main' into vitkyrka/discolog-filter

vitkyrka committed:
Merge remote-tracking branch 'origin/main' into vitkyrka/discolog-filter

 Conflicts:
	comp/core/workloadfilter/def/proto/filter.pb.go

vitkyrka committed:
Fix jmx

vitkyrka committed:
Merge remote-tracking branch 'origin/main' into vitkyrka/discolog-filter

 Conflicts:
	comp/core/workloadfilter/def/proto/filter.pb.go

vitkyrka committed:
Merge branch 'main' into vitkyrka/discolog-filter
